doc/tutorials/profiling.md

```diff
@@ -0,0 +1,80 @@
+# Profiling
+
+## Execution Time
+
+One of the primary resources in computing is execution time. To keep usage of this resource type low, it makes sense to profile code and check which code paths in a progamm take the longest time to execute. There exist various tools to handle this kind of profiling. For this tutorial we will use [Callgrind](http://valgrind.org/docs/manual/cl-manual.html), and the graphical frontend [KCacheGrind/QCacheGrind](https://kcachegrind.github.io/html/Home.html).
+
+### Choosing the Correct Build Type
+
+Since we want to improve the readability of the Callgrind output we choose a build type that includes debug symbols. The two obvious choices for the build type are:
+
+- `RelWithDebInfo` (optimized build with debug symbols), and
+- `Debug` (non-optimized build with debug symbols)
+
+. We use `Debug` here, which should provide the most detailed profiling information.
+
+### Disabling `dlclose` Calls
+
+For this tutorial we decided to profile the [YAy PEG][] plugin. Since Elektra loads plugin code via `dlopen` and Callgrind [does not support the function `dlclose` properly](https://stackoverflow.com/questions/16719395) we remove the `dlclose` calls in the file [`dl.c`](../../src/libs/loader/dl.c) temporarily. At the time of writing one option to do that is deleting
```

Comment:
What about ENABLE_DEBUG doing this? Or we let ENABLE_DEBUG not tamper with dlclose at all and rename it to ENABLE_ASSERT (see also above).

Reply:
> What about ENABLE_DEBUG doing this?

While a CMake option to enable this behavior would be nice, as far as I can remember disabling `dlclose` also helps to debug ASAN and Valgrind output, I think `ENABLE_DEBUG` should not introduce additional memory leaks.